server/src/services/formatting/onDocumentFormatting.ts

```diff
@@ -46,11 +47,17 @@ export function onDocumentFormatting(serverState: ServerState) {
               return { status: "invalid_argument", result: null };
           }
         } catch (error) {
-          serverState.logger.trace(
+          serverState.logger.info(
             `Error formatting document ${uri} with ${formatter}: ${error}`
           );
 
-          return { status: "internal_error", result: null };
+          if (error instanceof TimeoutError) {
+            return { status: "deadline_exceeded", result: null };
+          } else if (error instanceof ForgeResolveError) {
+            return { status: "failed_precondition", result: null };
+          } else {
+            return { status: "internal_error", result: null };
+          }
```

Comment:
```suggestion
          } else {
            serverState.logger.error(error);
            return { status: "internal_error", result: null };
          }
```

We effectively short-circuit the error handling of `trackTiming`; it would Sentry log the error. I don't think we want to do that for timeouts or failed preconditions - but what do we think of reporting internal errors?
I don't have a strong feeling either way.

Reply:
As of now, saving a solidity file with syntax errors would cause forge fmt to fail, and that would result in an internal_error. Should I handle the additional case where the program executed but exit code was non zero? and avoid reporting it to sentry ?

Reply:
Ah. No lets leave this reporting out completely then. Forge fmt failing on syntax errors is definitely not worth reporting.